### ISSUE TITLE
Expand/collapse layers API

### ISSUE THREAD

**Issue by hnbdr** [mapeditor/tiled#3997]
When working with maps with a large number of layers, a scripting API for expanding/collapsing GroupLayer and ObjectGroup would be very helpful.

As @bjorn suggested, this can be done by exposing the Layers and Objects views functions in the MapEditor interface.

**Comment by Oval17** [comment]
Hi @hnbdr @bjorn 
I'd like to work on this. I see PR #4009 made progress but has been open since July 2024 with some unresolved issues around segfault protection. 
Would it be appropriate if I work on top of that PR?

**Comment by bjorn** [comment]
@Oval17 Ah, thanks for bringing this PR to my attention again, but I really should provide some feedback to the original author and then we can give them some time to respond. If it turns out to be abandoned you could finish it, but I just had a look and it really seems like it's almost done.

**Comment by Oval17** [comment]
@bjorn Yes sure as I can see you have asked author regarding pr and reviewed the pr as well, will move accordingly.
Thanks

**Comment by bjorn** [comment]
@Oval17 Alright, it's been over a week now with no response from @PalaBeaveR, so I think it's alright if you continue their work. If you pull the branch from their fork and commit on top I can still easily credit you both when merging this.

**Comment by PalaBeaveR** [comment]
Very sorry. I meant to respond to this when i saw it, but lost track of it.
Feel free to take this over as i don't have the time or interest to work on
this right now.

**Comment by Oval17** [comment]
Hey @bjorn @PalaBeaveR 
Yes sure thanks for confirmation, will take over.

**Comment by PrateekSingh070** [comment]
Hi, I’m new to the project and interested in working on this issue.
I’m currently setting up the development environment and would like to try fixing this if it’s available.

**Comment by sam-coolshrestha** [comment]
Hi! I'd like to work on this issue if it's still available.

**Comment by bjorn** [comment]
@PrateekSingh070 @sam-coolshrestha I appreciate your enthusiasm but there is already a PR open for finishing this at #4379 so please look around for another issue to work on.

I realize this is currently the last open issue with explicit "good first issue" label, but any issue is a potentially good first one! You can also play around with Tiled to find things that could be improved. :-)
